opendevin/core/config.py

```diff
@@ -3,102 +3,292 @@
 import os
 import pathlib
 import platform
+from dataclasses import dataclass, field, fields
+from types import UnionType
+from typing import Any, ClassVar, get_args, get_origin
 
 import toml
 from dotenv import load_dotenv
 
-from opendevin.core.schema import ConfigType
-
 logger = logging.getLogger(__name__)
 
-DEFAULT_CONTAINER_IMAGE = 'ghcr.io/opendevin/sandbox'
-if os.getenv('OPEN_DEVIN_BUILD_VERSION'):
-    DEFAULT_CONTAINER_IMAGE += ':' + (os.getenv('OPEN_DEVIN_BUILD_VERSION') or '')
-else:
-    DEFAULT_CONTAINER_IMAGE += ':main'
-
 load_dotenv()
 
-DEFAULT_CONFIG: dict = {
-    ConfigType.LLM_API_KEY: None,
-    ConfigType.LLM_BASE_URL: None,
-    ConfigType.WORKSPACE_BASE: os.getcwd(),
-    ConfigType.WORKSPACE_MOUNT_PATH: None,
-    ConfigType.WORKSPACE_MOUNT_PATH_IN_SANDBOX: '/workspace',
-    ConfigType.WORKSPACE_MOUNT_REWRITE: None,
-    ConfigType.CACHE_DIR: '/tmp/cache',  # '/tmp/cache' is the default cache directory
-    ConfigType.LLM_MODEL: 'gpt-3.5-turbo-1106',
-    ConfigType.SANDBOX_CONTAINER_IMAGE: DEFAULT_CONTAINER_IMAGE,
-    ConfigType.RUN_AS_DEVIN: 'true',
-    ConfigType.LLM_EMBEDDING_MODEL: 'local',
-    ConfigType.LLM_EMBEDDING_BASE_URL: None,
-    ConfigType.LLM_EMBEDDING_DEPLOYMENT_NAME: None,
-    ConfigType.LLM_API_VERSION: None,
-    ConfigType.LLM_NUM_RETRIES: 5,
-    ConfigType.LLM_RETRY_MIN_WAIT: 3,
-    ConfigType.LLM_RETRY_MAX_WAIT: 60,
-    ConfigType.MAX_ITERATIONS: 100,
-    ConfigType.AGENT_MEMORY_MAX_THREADS: 2,
-    ConfigType.AGENT_MEMORY_ENABLED: False,
-    ConfigType.LLM_TIMEOUT: None,
-    ConfigType.LLM_MAX_RETURN_TOKENS: None,
-    ConfigType.LLM_TEMPERATURE: None,
-    ConfigType.LLM_TOP_P: None,
-    # GPT-4 pricing is $10 per 1M input tokens. Since tokenization happens on LLM side,
-    # we cannot easily count number of tokens, but we can count characters.
-    # Assuming 5 characters per token, 5 million is a reasonable default limit.
-    ConfigType.MAX_CHARS: 5_000_000,
-    ConfigType.AGENT: 'CodeActAgent',
-    ConfigType.E2B_API_KEY: '',
-    ConfigType.SANDBOX_TYPE: 'ssh',  # Can be 'ssh', 'exec', or 'e2b'
-    ConfigType.USE_HOST_NETWORK: 'false',
-    ConfigType.SSH_HOSTNAME: 'localhost',
-    ConfigType.DISABLE_COLOR: 'false',
-    ConfigType.SANDBOX_USER_ID: os.getuid() if hasattr(os, 'getuid') else None,
-    ConfigType.SANDBOX_TIMEOUT: 120,
-    ConfigType.GITHUB_TOKEN: None,
-    ConfigType.SANDBOX_USER_ID: None,
```

Comment:
this line was overriding 3 lines above. Looks like sandbox_user_id should be None.

Reply:
But this code says it should be (int) os.getuid() if hasattr(os, 'getuid') else 1000

https://github.com/OpenDevin/OpenDevin/blob/367632e36542b3490861c99df5aec6e2ffb40e93/opendevin/runtime/docker/ssh_box.py#L42